code/game/objects/items/mail.dm

```diff
@@ -332,3 +352,172 @@
 /obj/item/paper/fluff/junkmail_generic/Initialize(mapload)
 	default_raw_text = pick(GLOB.junkmail_messages)
 	return ..()
+
+/obj/item/mail/traitor
+	var/armed = FALSE
+	var/datum/weakref/made_by
+	// If somehow mind will disappear, admins will still have info about creator
+	var/made_by_cached_name
+	var/made_by_cached_ckey
+	goodie_count = 0
+
+/obj/item/mail/traitor/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/traitor/after_unwrap(mob/user)
+	user.temporarilyRemoveItemFromInventory(src, force = TRUE)
+	playsound(loc, 'sound/items/poster_ripped.ogg', vol = 50, vary = TRUE)
+	for(var/obj/item/stuff in contents) // Mail and envelope actually can have more than 1 item.
+		if(user.put_in_hands(stuff) && armed)
+			log_bomber(user, "opened armed mail made by [made_by_cached_name] ([made_by_cached_ckey]), activating", stuff)
+			INVOKE_ASYNC(stuff, TYPE_PROC_REF(/obj/item/mail/traitor, attack_self), user)
+	qdel(src)
+
+/obj/item/mail/traitor/multitool_act(mob/living/user, obj/item/tool)
+	if(armed == FALSE || user.get_inactive_held_item() != src)
+		. = ..()
+	else
+		if(user.mind == made_by.resolve())
```

Comment:
Okay, I get why you want to hold onto this now, that's fine.

```suggestion
		if(IS_WEAKREF_OF(user.mind, made_by))
```

This will not pass if mind is null, which is correct

Reply:
understood